src/golang/lib/lambda/utils.go

```diff
@@ -23,15 +23,49 @@ type EcrAuth struct {
 	ProxyEndpoint string
 }
 
-func CreateLambdaFunction(functionType LambdaFunctionType, roleArn string) error {
-	// For each lambda function we create, we take the following steps:
-	// 1. Pull the image from the public ECR repository.
-	// 2. Create the private ECR repo if it doesn't exist
-	// 3. Get the ECR auth token and log in the docker client.
-	// 4. Push the image to the private ECR repo
-	// 5. Create the lambda function using the private ECR repo as the code.
+func AuthenticateDockerToECR() error {
+	// Authenticate ECR client.
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
 
-	lambdaImageUri, userRepoName, err := mapFunctionType(functionType)
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	ecrSvc := ecr.New(sess)
+
+	token, err := ecrSvc.GetAuthorizationToken(&ecr.GetAuthorizationTokenInput{})
+	if err != nil {
+		return errors.Wrap(err, "Unable to get authorization token.")
+	}
+	auth, err := extractToken(*token.AuthorizationData[0].AuthorizationToken, *token.AuthorizationData[0].ProxyEndpoint)
+	if err != nil {
+		return errors.Wrap(err, "Unable to extract username and password.")
+	}
+
+	cmd := exec.Command(
+		"docker",
+		"login",
+		"--username",
+		auth.Username,
+		"--password",
+		auth.Password,
+		auth.ProxyEndpoint,
+	)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	err = cmd.Run()
+	if err != nil {
+		log.Info(stdout.String())
+		log.Info(stderr.String())
+		return errors.Wrap(err, "Unable to authenticate docker client to ECR.")
+	}
+	return nil
+}
+
+func PullImageFromECR(functionType LambdaFunctionType) error {
+	// Pull the Image from public ECR Library.
```

Comment:
Will the images be pulled onto the server machine? Should we worry about deleting the images locally after they are pushed? My impression was that they took up a ton of space. cc: @hsubbaraj-spiral